torch/testing/_internal/common_modules.py

```diff
@@ -189,6 +189,36 @@ def module_inputs_torch_nn_Linear(module_info, device, dtype, requires_grad, **k
     return module_inputs
 
 
+def module_inputs_torch_nn_Bilinear(module_info, device, dtype, requires_grad, **kwargs):
+    make_input = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+    
+    def bilinear_reference_fn(m, p, x1, x2, bias=True):
+        intermediate = torch.einsum('bk,ikj->bij', x1, p[0])
+        result = torch.einsum('bij,bj->bi', intermediate, x2)
```

Comment:
Looks like you may be able to use `torch.einsum('bn,anm,bm->ba', l, A, r)` to do this in one step? Check out the final example in the [einsum docs](https://pytorch.org/docs/stable/generated/torch.einsum.html)